workshops/rust_text_game/README.md

```diff
@@ -0,0 +1,405 @@
+ 
+Now run your program again. Instead of `Hello, World`, you should see what you wrote above
+
+### Part 4.2: User input
+
+A game needs to give the ability the option to interact with it and affect its course.
+
+... After all, [it's what makes a Subaru a Subaru!](https://www.youtube.com/watch?v=aYVoAioo8Ww)
+
+So, we'll write a new function that will accept read the player's answer and react accordingly.
+
+For purposes of simplicity, the player will only be supposed to answer with a word starting with "Y", which includes words such as 'Yes', 'Yep', 'Yeah', or 'Yellow' (as a funny unintended consequence), or, likewise, "N".
+
+`src/prompts.rs`:
+```rust
+use std::io;
+
+fn prompt() -> bool {
+    let mut input = String::new();
+    io::stdin().read_line(&mut input).unwrap();
+
+    return input.to_ascii_lowercase().starts_with("y");
+}
+```
+
+Now, I know that this is a lot to take in, but don't fret! Let's just go through what's happening here once more, line-by-line:
+
+   - `use std::io;`
+    
+    Importing libraries allows us to use code either installed by cargo, or bundled with Rust (standard library). By importing `std::io` (where `std` means standard library aka included with Rust), we gain access to more functions that allow us to perform "input/output" operations (like `println!()`, but more advanced!).
+
+- `fn prompt() -> bool {`
+
+    All functions are meant to return some output, which is either visible to the player, to other programs or parts of the same program. In this case, we define a new function called `prompt()`, which will return a value known as a `bool`, which stands for "boolean". Boolean data types can only accept two values, which are denoted with the words `true` or `false` in the Rust programming language. They're named after the mathematician George Boole and are possibly the simplest instance of [Boolean algebra](https://en.wikipedia.org/wiki/Boolean_algebra) in the field of computer science.
+
+- `let mut input = String::new();`
+
+    Here, we create a *mutable* `String`. *Mutability* defines whether you can change the value of a variable. Since we're *initializing* a new string (which means that we declare that the variable `input` exists and is a `String`, but there haven't been any values that have been *declared* to it yet.)
+
+- `io::stdin().read_line(&mut input).unwrap();`
+
+    Here, using the `io` library again, we capture whatever the player is typing until they hit `Enter` inside the `input` variable.
+
+- `return input.to_ascii_lowercase().starts_with("y");`
+
+    And finally, we check if an all-lowercase counterpart of the input's value starts with `y`. If it does, it means the input was yes (or similar), and returns `true`. If it doesn't, it's safe to assume that the player replied negatively. Therefore, we return the value `false`.
```

Comment:
```suggestion
    Finally, we must check if the `input` variable starts with either a lowercase or capital `y`. We're passing what's called a closure to `starts_with`; you can think of it as a condensed function that you can pass as a parameter. Here, this closure takes the parameter `ch`, which is the character you're matching, and uses the `matches!` macro to check if it matches either "y" or (`|`) "Y". This will return the boolean `true` if it finds the match, and `false` if it doesn't.
```